### PR TITLE
Should fix #3600 : added 2 more CV-related tests

### LINKED ISSUE #3600: Hammer: Automate few more tests around composite content-view
- Create a composite content-view with --component-ids option
- Update a composite content-view with --component-ids option
- Remove content-view version from composite-view

### PR DESCRIPTION
**Pull request by helgie**
Implemented 

* Update a composite content-view with --component-ids option
* Remove content-view version from composite-view

Test results: 

helgie$ py.test tests/foreman/cli/test_contentview.py -k test_positive_remove_version_by_id_from_composite
========= test session starts =========
platform darwin -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/helgie/robottelo, inifile: 
plugins: cov-2.3.0, xdist-1.14
collected 52 items 
tests/foreman/cli/test_contentview.py .
======= 51 tests deselected by '-k test_positive_remove_version_by_id_from_composite' ========== 1 passed, 51 deselected in 96.82 seconds 

helgie$ py.test tests/foreman/cli/test_contentview.py -k test_positive_update_composite_with_component_ids
========= test session starts ==========
platform darwin -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/helgie/robottelo, inifile: 
plugins: cov-2.3.0, xdist-1.14
collected 52 items 

tests/foreman/cli/test_contentview.py .

=== 51 tests deselected by '-ktest_positive_update_composite_with_component_ids' 
=========1 passed, 51 deselected in 91.86 seconds ========

> fix #3600

### CODE REVIEW

File: tests/foreman/cli/test_contentview.py [hunk 1]
```diff
@@ -647,6 +647,37 @@ def test_positive_create_composite(self):
...
+        @assert: Composite content view info output does not contain any values
+
+        @CaseLevel: Integration
+        """
+        # Create CV
+        comp_cv = make_content_view({
+            'composite': True,
+            'organization-id': self.org['id']
+        })
+        ContentView.publish({u'id': comp_cv['id']})
```
Comment (sghai) [inline tests/foreman/cli/test_contentview.py]: Looks like we are publishing a blank( without any CV) composite content-view and removing that version ?
I think valid test would be like:
- create a composite cv
- add a CV( with some contents) to composite CV created in step1
- publish the composite cv
- remove published version

If there is already a test based on above steps. Then I'm fine with this test too but with updated docstring as it doesn't contain any contents or CV

File: tests/foreman/cli/test_contentview.py [hunk 2]
```diff
@@ -682,6 +713,45 @@ def test_positive_create_composite_with_component_ids(self):
...
+
+        @id: e6106ff6-c526-40f2-bdc0-ae291f7b267e
+
+        @assert: Composite content view component ids are similar to the
+        nested content view versions ids
+
+        @CaseLevel: Integration
+        """
+        # Create a CV to add to the composite one
+        inserted_cv = make_content_view({u'organization-id': self.org['id']})
```
Comment (sghai) [inline tests/foreman/cli/test_contentview.py]: `can we rename `inserted_cv` to some other name ? may be leave it as  `cv` or `cv1` ?

File: tests/foreman/cli/test_contentview.py [hunk 3]
```diff
@@ -682,6 +713,45 @@ def test_positive_create_composite_with_component_ids(self):
...
+        # Publish a new version of the CV
+        ContentView.publish({u'id': inserted_cv['id']})
+        new_cv = ContentView.info({u'id': inserted_cv['id']})
+        # Let us now store the version ids
+        component_ids = new_cv['versions'][0]['id']
+        # Create CV
```
Comment (sghai) [inline tests/foreman/cli/test_contentview.py]: # Create composite CV ?

**Comment by sghai** [comment]
ACK pending few comments

**Comment by sghai** [comment]
ACK.. Looks good. Please add test results once more.

**Comment by helgie** [comment]
Some fresh test results: 

helgie$ py.test tests/foreman/cli/test_contentview.py -k test_positive_update_composite_with_component_ids
================ test session starts ======================
platform darwin -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/helgie/robottelo, inifile: 
plugins: cov-2.3.0, xdist-1.14
collected 52 items 

tests/foreman/cli/test_contentview.py .
============= 51 tests deselected by '-ktest_positive_update_composite_with_component_ids' ===========
======================== 1 passed, 51 deselected in 102.66 seconds ===================================

helgie$ py.test tests/foreman/cli/test_contentview.py -k test_positive_remove_version_by_id_from_composite
====================== test session starts =============================================================
platform darwin -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/helgie/robottelo, inifile: 
plugins: cov-2.3.0, xdist-1.14
collected 52 items 

tests/foreman/cli/test_contentview.py .

======= 51 tests deselected by '-ktest_positive_remove_version_by_id_from_composite' =====================
=========== 1 passed, 51 deselected in 191.96 seconds ====================================================

File: tests/foreman/cli/test_contentview.py [hunk 4]
```diff
@@ -647,6 +647,55 @@ def test_positive_create_composite(self):
...
+        })
+        # Publish a new version of CV
+        ContentView.publish({u'id': new_cv['id']})
+        # Get the CV info
+        new_cv = ContentView.info({u'id': new_cv['id']})
+        # Create a composite CV
+        comp_cv = make_content_view({
+            'composite': True,
+            'organization-id': self.org['id'],
+            '--component-ids': new_cv['versions'][0]['id']
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: `--component-ids?`
Can't we use `component-ids` ?

File: tests/foreman/cli/test_contentview.py [hunk 5]
```diff
@@ -647,6 +647,55 @@ def test_positive_create_composite(self):
...
+        """Create a composite content view and remove its content version by id
+
+        @id: 0ff675d0-45d6-4f15-9e84-3b5ce98ce7de
+
+        @assert: Composite content view info output does not contain any values
+
+        @CaseLevel: Integration
+        """
+        # Create new organization, product and repository
+        new_org = make_org({u'name': gen_alphanumeric()})
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: 1) Do we really need new organization and product? Can't we re-use them from setup section?
2) Do we need to have product and repository for these tests?

**Comment by oshtaier** [comment]
ACK

**Pull request merged**